google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/JsonToProtoMessage.java

```diff
@@ -182,16 +192,32 @@ public DynamicMessage convertToProtoMessage(
    * @param ignoreUnknownFields allows unknown fields in JSON input to be ignored.
    * @throws IllegalArgumentException when JSON data is not compatible with proto descriptor.
    */
-  public DynamicMessage convertToProtoMessage(
```

Comment:
Let's add a new method instead of removing this old one, in case there are someone who is already using it. And also put some comments on recommending using the new API.

Reply:
IMO, this class is exposing terrible APIs

It's exposing 
public convertToProtoMessage(..., JSONObject object, ...)
public convertToProtoMessage(..., Object object, ...)
and bunch of other public APIs

The first two APIs are already bad enough, one of it is overriding the interface while the other is not
They are doing type infer based on whether it's jsonObject or normal object 
it's so terrible..
After exposing another method, I really doubt whether anyone would every use it..

Reply:
Yes, but we don't know if they are using it. We could provide examples of implementation in the comment since public convertToProtoMessage(..., JSONObject object, ...) can always directly use public convertToProtoMessage(..., Object object, ...) underneath. Also should the second method be JSONArray?

Reply:
They exposed both: 

 Descriptor protoSchema, TableSchema tableSchema, Object json, boolean ignoreUnknownFields) 
and  Descriptor protoSchema, TableSchema tableSchema, Json json, boolean ignoreUnknownFields) 

The former one implements the interface while the later one does not..